packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -280,11 +282,23 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         this._attachState = this.containerRuntime.attachState !== AttachState.Detached && this.existing ?
             this.containerRuntime.attachState : AttachState.Detached;
 
+        /**
+         * The existing flag can be true in two conditions:
+         * 1. It's a local data store that is created when a detached container is rehydrated. In this case, the data
+         *    store is locally visible because the snapshot it is loaded from contains locally visible data stores only.
+         * 2. It's a remote data store that is created when an attached container is loaded is loaded from snapshot or
+         *    when an attach op comes in. In both these cases, the data store is already globally visible.
+         */
+        if (existing) {
```

Comment:
I don't think it existing vs this.existing matter in the constructor.
To pass in visibility state here, we need to teach visibility state to ContainerRuntime / DataStores. We can (and probably should) do that but I don't want to do it in this PR. The `existing` flag drives other things in layers below data store context as well so we cannot easily remove it.

Reply:
Here's a techdebt issue https://github.com/microsoft/FluidFramework/issues/9646

Reply:
You should add that you are talking about the data store context layer. This is because existing flag is in other layers too. Also, visibility state is added by this PR which isn't merged yet so you should mention that you are talking about the state as is being added by this PR.
Andrei had worked on removing existing state from a bunch of places so might be worth talking to him about this.